docs/source/chainermn/model_parallel/example4_ensemble.rst

```diff
@@ -0,0 +1,43 @@
+Example 4: Ensemble
+===================
+
+Averaging ensemble is one application which collective communications can effectively be applied to.
+
+.. figure:: ../../../image/model_parallel/averaging.png
+    :align: center
+
+The following wrapper makes model parallel averaging ensemble easier::
+
+    class Averaging(chainer.Chain):
+        def __init__(self, comm, block):
+            super(Averaging, self).__init__()
+            self.comm = comm
+            with self.init_scope():
+                self.block = block
+
+        def __call__(self, x):
+            y = self.block(x)
+    
+            if not chainer.config.train:
+                y = chainermn.functions.allgather(self.comm, y)
+                y = F.stack(y, axis=0)
+                y = F.average(y, axis=0)
+
+            return y
+
+Then, any links wrapped by ``Averaging`` are ready to be parallelized and averaged::
+
+    class Model(chainer.Chain):
+        def __init__(self, comm):
+            super(Model, self).__init__()
+            self.comm = comm
+            with self.init_scope():
+                self.l1 = L.Linear(d0, d1)
+                self.l2 = L.Linear(d1, d2)
+                self.l3 = Averaging(self.comm, L.Linear(d2, d3))
+
+        def __call__(self, x):
+            h = F.relu(self.l1(x))
+            h = F.relu(self.l2(h))
+            y = F.relu(self.l3(h))
+            return y
```

Comment:
I think same input must be set to ensembled models. So it would be more narrative if there's description and example using `MultiNodeIterator`, or just we don't need `scatter_dataset`? At least I think we'd better have description on it.

Reply:
Also the behaviour of averaged model from input/output perspective should be compatible with original models, right? Would be nice there's description on it, say like "can be replaced easily..." or "compatible".